### PR TITLE
Add Bearer token support to FF Auth

### PR DESCRIPTION
**Pull request by hardillb**
part of https://github.com/FlowFuse/flowfuse/issues/2964
## Description

Allow HTTP Bearer tokens for Node-RED HTTP endpoints when using FF auth

image

requires https://github.com/FlowFuse/nr-launcher/pull/212

## Related Issue(s)

https://github.com/FlowFuse/flowfuse/issues/2964

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: frontend/src/pages/instance/Settings/dialogs/TokenDialog.vue [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    <ff-dialog ref="dialog" data-el="add-token-dialog" :header="(token ? 'Edit' : 'Add') + ' Token'" :confirm-label="token ? 'Save' : 'Create'" :disablePrimary="disableConfirm" @confirm="confirm()">
+        <template #default>
+            <form class="space-y-4" @submit.prevent>
+                <FormRow v-model="input.name" data-form="token-name" :disabled="edit">
+                    Token name
+                </FormRow>
+                <ff-checkbox v-model="input.expires" data-form="expiry-toggle" label="Add Expiry Date" />
+                <FormRow v-model="input.expiresAt" data-form="token-expiry" type="date" :disabled="!input.expires">
+                    Expires
+                    <!-- <template v-slot:description>Expires</template> -->
```
Comment (Steve-Mcl) [inline frontend/src/pages/instance/Settings/dialogs/TokenDialog.vue:11]:
```suggestion
```

**Review by Steve-Mcl — COMMENTED** [review]
Tested locally in conjunction with current and associated nr-launcher branch. Working as expected however there are a few minor points that might want to be addressed:

1. The token creation dialog permits a user to chose a date in the past. I have verified the token is not useable but ideally we would add a `min` value on the input element to inhibit this. Alternatively, client side logic that toasts a message?.
2. A token created for todays date does not work. Is that expected? A user may think they are creating a token for the remainder of the day (I did) but looking at the database, I can see the expire time is set as `yyyy/mm/dd 00:00:00` (i.e. this morning 00:00am)
3. There is no update to the docs for this. A short nod to this feature in `docs/user/instance-settings.md#Security` would be enough IMHO. 
   * Nice to have: Include a `cURL` example. Ideally this would by a copyable/usable code div on the token creation dialog, but at minimum a code example in the docs so the users know not to base64 encode the token for example?

**Comment by hardillb** [comment]
>  1. The token creation dialog permits a user to chose a date in the past. I have verified the token is not useable but ideally we would add a min value on the input element to inhibit this. Alternatively, client side logic that toasts a message?

The token will be removed as soon as it is tested after the expiry date.

If you know how to add a min value to a the FF-Form-Row to make it add `min` attribute please share:

```
<FormRow v-model="input.expiresAt" data-form="token-expiry" type="date" min="{ new Date().toJSON().split('T')[0] }" :disabled="!input.expires">
```

doesn't work
.
  > 2. A token created for todays date does not work. Is that expected? A user may think they are creating a token for the remainder of the day (I did) but looking at the database, I can see the expire time is set as yyyy/mm/dd 00:00:00 (i.e. this morning 00:00am)

Tokens are valid until the expiry date, this is working as expected

   > 3. There is no update to the docs for this. A short nod to this feature in docs/user/instance-settings.md#Security would be enough IMHO.
        Nice to have: Include a cURL example. Ideally this would by a copyable/usable code div on the token creation dialog, but at minimum a code example in the docs so the users know not to base64 encode the token for example?

Docs I can add.

**Comment by Steve-Mcl** [comment]
@hardlib, I added date validation in commit [5d23583](https://github.com/FlowFuse/flowfuse/pull/3535/commits/5d235830b182fed6823cd912ac8c3cbc663d8704)


chrome_xsnDtj1Y2a

**Pull request merged**